### ISSUE TITLE
[Typo]: Sentence out of place?

### ISSUE THREAD

**Issue by txbm** [reactjs/react.dev#6881]
### Summary

The highlighted sentence seems out of place. It does not match the context of the page.

### Page

https://react.dev/reference/rules/react-calls-components-and-hooks

### Details

![Screenshot 2024-05-18 at 6 53 15 PM](https://github.com/reactjs/react.dev/assets/102656/c08ba9b6-ed96-4d5c-b897-a933c07b98be)

**Comment by rickhanlonii** [comment]
It's not out of place but it could be explained better. The missing context is that this pattern of passing hooks as props is commonly done as a way to do dependency injection so you can replace the hook with a mock in testing environments. Maybe we don't need to include that, but if we want to keep it we should provide more of this context.

cc @poteto

**Comment by poteto** [comment]
Yeah I was trying to think of examples where people might reach for dependency injection and testing seemed like the most common scenario. Adding some context about this would be great, @txbm would you be interested in opening a PR?

**Comment by CuriousCoder00** [comment]
![image](https://github.com/reactjs/react.dev/assets/103230903/3d5358ce-cad8-48cb-8eb7-d8cb2e6c8e7b)
